src/service/video-manager-impl.js

```diff
@@ -418,6 +426,78 @@ class VideoEntry {
   }
 
   /**
+   * Gets the provided metadata and fills in missing fields
+   * @return {!../video-interface.VideoMetaDef}
+   * @private
+   */
+  getMetaData_() {
+    let metaData = this.video.metaData;
+    if (!metaData) {
+      metaData = {};
+    }
+    if (!metaData.artist) {
+      metaData.artist = 'No artist';
+    }
+    if (!metaData.posterUrl) {
+      metaData.posterUrl = this.video.element.getAttribute('poster')
+                           || this.internalElement_.getAttribute('poster')
+                           || this.getDefaultPoster_();
+    }
+    if (!metaData.title) {
+      metaData.title = this.video.element.getAttribute('title')
+                       || this.video.element.getAttribute('aria-label')
+                       || this.internalElement_.getAttribute('title')
+                       || this.internalElement_.getAttribute('aria-label')
+                       || this.ampdoc_.win.document.title;
+    }
+    if (!metaData.album) {
+      metaData.album = 'No album';
+    }
+    return metaData;
+  }
+
+  /**
+   * Gets the provided metadata and fills in missing fields
+   * @private
+   */
+  mediaSessionUpdate_() {
+    const win = this.ampdoc_.win;
+    const navigator = win.navigator;
+    if ('mediaSession' in navigator && win.MediaMetadata) {
+
+      navigator.mediaSession.metadata = new win.MediaMetadata({
+        title: this.metaData_.title,
+        artist: this.metaData_.artist,
+        album: this.metaData_.album,
+        artwork: [
+          {
+            src: this.metaData_.posterUrl,
+            sizes: '512x512',
+            type: 'image/png',
```

Comment:
Not mandatory, just checked. Only useful when having multiple images for different sizes.